### PR TITLE
feat: added basic swap ui flow

### PR DESCRIPTION
**Pull request by addegbenga**
As part of this [PR](https://github.com/AFK-AlignedFamKernel/afk_monorepo/issues/167) I added a basic swap component with initial functionality:

- Users can select two token pairs and view the estimated value in dollars.
- Added a context to trigger the swap functionality from anywhere in the app.
- Created custom hooks (`avnu hooks`) to support the swap process.
- Ability to swap tokens on layer2

### CODE REVIEW

File: apps/mobile/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "@dynamic-labs/react-native-extension": "4.0.0-alpha.8",
     "@dynamic-labs/utils": "^3.3.0",
     "@dynamic-labs/viem-extension": "4.0.0-alpha.8",
+    "@ethersproject/units": "^5.7.0",
```
Comment (MSghais) [inline apps/mobile/package.json:40]: Can we delete a lib? We already have few utils and others lib doing it please

File: apps/mobile/src/modules/Swap/index.tsx [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+      },
+      {
+        async onSuccess(data) {
+          // Find the swap call
+          const swapCall = data.calls.find(
+            (call) => call.entrypoint === 'swap' || call.entrypoint === 'multi_route_swap',
+          );
+
+          if (swapCall) {
+            sendTransaction(
```
Comment (MSghais) [inline apps/mobile/src/modules/Swap/index.tsx:167]: Can you change it with a Starknet account and real calldata of the avnu?

File: apps/mobile/src/modules/Swap/index.tsx [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+interface Token {
+  symbol: string;
+  l2_token_address: string;
+  decimals: number;
+  logo_url: string;
+}
+
+export default function TokenSwapView({showHeader = false}: {showHeader?: boolean}) {
+  const {showSwap} = useSwapModal();
+  const {sendTransaction} = useSendTransaction();
```
Comment (MSghais) [inline apps/mobile/src/modules/Swap/index.tsx:29]: AVNU is a Starknet DEX, we have hook for Starknet already and modal to connect Argent/Bravoos account

**Review by MSghais — CHANGES_REQUESTED** [review]

**Comment by MSghais** [comment]
@addegbenga do you want to finish the Starknet account execution on this PR?

Otherwise I can merge it as it and we create another issue with the remaining tasks, let me know

**Comment by addegbenga** [comment]
Yeah I will work on the rest): Thanks

**Review by MSghais — APPROVED** [review]
LFG

**Pull request merged**